pyiron_atomistics/calphy/job.py

```diff
@@ -720,39 +733,45 @@ def collect_general_output(self):
         Returns:
             None
         """
-        if self._data is not None:
-            # solid liquid specific outputs
-            if "spring_constant" in self._data["average"].keys():
-                self.output["spring_constant"] = self._data["average"][
-                    "spring_constant"
-                ]
-            if "density" in self._data["average"].keys():
-                self.output["atomic_density"] = self._data["average"]["density"]
-            self.output["atomic_volume"] = self._data["average"]["vol_atom"]
+        reportfile = os.path.join(self.working_directory, "report.yaml")
+
+        if os.path.exists(reportfile):
+            with open(reportfile, "r") as fin:
+                data = yaml.safe_load(fin)
+
+            concentration = data["input"]["concentration"].split()
+            concentration = [float(x) for x in concentration]
+
+            if "spring_constant" in data["average"].keys():
+                self.output["spring_constant"] = data["average"]["spring_constant"]
+            if "density" in data["average"].keys():
+                self.output["atomic_density"] = data["average"]["density"]
+            self.output["atomic_volume"] = data["average"]["vol_atom"]
 
             # main results from mode fe
             self.output["temperature"] = self.input.temperature
             self.output["pressure"] = self.input.pressure
-            self.output["energy_free"] = self._data["results"]["free_energy"]
-            self.output["energy_free_error"] = self._data["results"]["error"]
-            self.output["energy_free_harmonic_reference"] = self._data["results"][
+            self.output["energy_free"] = data["results"]["free_energy"]
+            self.output["energy_free_error"] = data["results"]["error"]
+            self.output["energy_free_harmonic_reference"] = data["results"][
                 "reference_system"
             ]
-            self.output["energy_work"] = self._data["results"]["work"]
-            self.output["energy_pressure"] = self._data["results"]["pv"]
+            self.output["energy_work"] = data["results"]["work"]
+            self.output["energy_pressure"] = data["results"]["pv"]
 
             # collect ediffs and so on
-            f_ediff, b_ediff, flambda, blambda = self._collect_ediff()
+            f_ediff, b_ediff, flambda, blambda = self._collect_ediff(concentration)
             self.output["fe/forward/energy_diff"] = list(f_ediff)
             self.output["fe/backward/energy_diff"] = list(b_ediff)
             self.output["fe/forward/lambda"] = list(flambda)
             self.output["fe/backward/lambda"] = list(blambda)
 
             # get final structure
-            traj = PyscalTrajectory(
-                os.path.join(self.working_directory, "conf.equilibration.dump")
+            aseobj = read(
```

Comment:
```suggestion
            aseobj = self.project.create.structure.read(
```

This should save you the `ase_to_pyiron` call below.